### ISSUE TITLE
Canonicalize host

### ISSUE THREAD

**Issue by luqmana** [oxidecomputer/oxide.rs#163]
I set the `OXIDE_HOST` environment variable to `https://pluto.oxide-preview.com` (note lack of trailing slash) and tried to run the cli but ran into a panic:

```console
$ oxide current-user ssh-key create [...]
thread 'main' panicked at 'called `Result::unwrap()` on an `Err` value: host https://pluto.oxide-preview.com not found', cli/src/main.rs:121:40
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
```

My `~/.config/oxide/hosts.toml` has multiple entries (hence why I explcitily was selecting with the env var):
```toml
["https://venus.oxide-preview.com/"]
token = "..."
user = "..."

["https://pluto.oxide-preview.com/"]
token = "..."
user = "..."
```

Note those have the trailing slashes.

**Comment by askfongjojo** [comment]
Bumped this to FCS as it's likely some users will stumble upon this and not know how to resolve or work around it.

**Comment by askfongjojo** [comment]
No longer ran into this error on commit `5756510ac1a272cb33c1d444750336037bd49b96`. We can perhaps close this?

**Comment by askfongjojo** [comment]
I was wrong about the status of this issue. It doesn't happen when the `OXIDE_TOKEN` environment variable is already set but it shows up when the variable is unset.
```
$ export OXIDE_HOST=https://oxide.sys.rack2.eng.oxide.computer/
$ unset OXIDE_TOKEN
$ oxide auth login --host $OXIDE_HOST
You're already logged into https://oxide.sys.rack2.eng.oxide.computer/
You're already logged into https://oxide.sys.rack2.eng.oxide.computer/
Do you want to re-authenticate? no

$ export OXIDE_HOST=https://oxide.sys.rack2.eng.oxide.computer
$ oxide auth login --host $OXIDE_HOST
thread 'tokio-runtime-worker' panicked at 'called `Result::unwrap()` on an `Err` value: host https://oxide.sys.rack2.eng.oxide.computer not found', cli/src/cli_builder.rs:232:40
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
thread 'main' panicked at 'called `Result::unwrap()` on an `Err` value: JoinError::Panic(Id(9), ...)', cli/src/main.rs:59:10
```

**Comment by david-crespo** [comment]
This is still an issue IMO, though maybe in different form than stated above. I would state it in terms of a mismatch between what is required for `oxide auth login` to tell you you're already logged in (it is more flexible) and what `OXIDE_HOST` has to be in order for requests to actually work.

```
$ oxide auth login --host oxide.sys.rack2.eng.oxide.computer
You're already logged into https://oxide.sys.rack2.eng.oxide.computer/
You're already logged into https://oxide.sys.rack2.eng.oxide.computer/
Do you want to re-authenticate? no
 ~/oxide/oxide.rs  [main]
$ oxide auth login --host https://oxide.sys.rack2.eng.oxide.computer
You're already logged into https://oxide.sys.rack2.eng.oxide.computer/
You're already logged into https://oxide.sys.rack2.eng.oxide.computer/
Do you want to re-authenticate? no
 ~/oxide/oxide.rs  [main]
$ oxide auth login --host https://oxide.sys.rack2.eng.oxide.computer/
You're already logged into https://oxide.sys.rack2.eng.oxide.computer/
You're already logged into https://oxide.sys.rack2.eng.oxide.computer/
Do you want to re-authenticate? no
 ~/oxide/oxide.rs  [main]
$ OXIDE_HOST='oxide.sys.rack2.eng.oxide.computer' oxide ip-pool list
$OXIDE_HOST is set, but oxide.sys.rack2.eng.oxide.computer has no corresponding token.
Login without $OXIDE_HOST set or set $OXIDE_TOKEN.
 ~/oxide/oxide.rs  [main]
$ OXIDE_HOST='https://oxide.sys.rack2.eng.oxide.computer' oxide ip-pool list
$OXIDE_HOST is set, but https://oxide.sys.rack2.eng.oxide.computer has no corresponding token.
Login without $OXIDE_HOST set or set $OXIDE_TOKEN.
 ~/oxide/oxide.rs  [main]
$ OXIDE_HOST='https://oxide.sys.rack2.eng.oxide.computer/' oxide ip-pool list
# successfully prints IP pools...
```

You can see above that `oxide auth login` does not care if there is an protocol or a trailing slash on the host when you try to log in, but the actual API calls do. The failure mode for me as a user is that I run `oxide auth login` to check if I'm logged in:

```
$ oxide auth login --host oxide.sys.rack2.eng.oxide.computer
You're already logged into https://oxide.sys.rack2.eng.oxide.computer/
You're already logged into https://oxide.sys.rack2.eng.oxide.computer/
Do you want to re-authenticate? no
```

And it tells me I am! So I say ok great, I'll just set `OXIDE_HOST=oxide.sys.rack2.eng.oxide.computer` and be on my way. 

```
$ OXIDE_HOST='oxide.sys.rack2.eng.oxide.computer' oxide ip-pool list
$OXIDE_HOST is set, but oxide.sys.rack2.eng.oxide.computer has no corresponding token.
Login without $OXIDE_HOST set or set $OXIDE_TOKEN.
```

But that doesn't work! So now I have to go through this back and forth to figure out what the working version of the host is. And (somewhat strangely IMO) it requires a trailing slash? ~~Or maybe it only does because that's what I used when I initially authed? Or maybe because I have an old auth config?~~ I reauthed and it behaves the same way.

**Comment by ahl** [comment]
I believe the need for this has either been addressed or obviated by #727
